include/assert.h

```diff
@@ -118,21 +118,22 @@
  *
  * Reference link:
  * https://pubs.opengroup.org/onlinepubs/009695399/basedefs/assert.h.html
- *
- * ASSERT/VERIFY is a non-standard interface, implemented using internal
- *
  */
 
 #ifdef NDEBUG
 #  define assert(f) ((void)0)
-#  define ASSERT(f) ((void)(1 || (f)))
-#  define VERIFY(f) ((void)(1 || (f)))
 #else
 #  define assert(f) _ASSERT(f, __ASSERT_FILE__, __ASSERT_LINE__)
-#  define ASSERT(f) _ASSERT(f, __ASSERT_FILE__, __ASSERT_LINE__)
-#  define VERIFY(f) _VERIFY(f, __ASSERT_FILE__, __ASSERT_LINE__)
 #endif
 
+/* ASSERT/VERIFY are NuttX-specific APIs.
+ * They are always enabled, regardless of NDEBUG/CONFIG_DEBUG_ASSERTIONS.
+ * The argument is evaluated exactly once.
+ */
+
+#define ASSERT(f) _ASSERT(f, __ASSERT_FILE__, __ASSERT_LINE__)
```

Comment:
but the change will slow down the execution even in release build.

Reply:
if the small speedup is important for you, i'd suggest you to investigate and replace ASSERT with DIAGASSERT one by one, instead of making a false assumption on how these APIs are used and break things.

Reply:
well, as far as i know, it has been that way.
but recently changed, breaking things.
this is a fix for the regression.